tests/crowdsourcing/tasks/turn_annotations_static/test_analysis.py

```diff
@@ -0,0 +1,172 @@
+                    "bucket_4": True,
+                    "none_all_good": False,
+                },
+            }
+            with open(temp_gold_annotations_path, 'w') as f:
+                json.dump(gold_annotations, f)
+
+            # Run compilation of results
+            opt = {
+                'results_folders': os.path.join(
+                    os.path.dirname(os.path.abspath(__file__)), 'analysis_samples'
+                ),
+                'output_folder': tmpdir,
+                'onboarding_in_flight_data_file': os.path.join(
+                    analysis_config_folder, 'onboarding_in_flight.jsonl'
+                ),
+                'gold_annotations_file': temp_gold_annotations_path,
+            }
+            with testing_utils.capture_output() as output:
+                compiler = TurnAnnotationsStaticResultsCompiler(opt)
+                compiler.NUM_SUBTASKS = 3
+                compiler.NUM_ANNOTATIONS = 3
+                compiler.compile_results()
+                actual_stdout = output.getvalue()
+
+            # Check the output against what it should be
+            desired_stdout = f"""\
+Got 4 folders to read.
+Average task completion time (seconds) was: 187.5
+Returning master dataframe with 48 annotations.
+Dropped 1 inconsistently annotated utterances (none_all_good and a problem bucket). Now have 7 utterances.
+Removed 1 that did not have annotations by 3 workers. 6 annotations remaining.
+summed_df has length 2; bot_only_df: 4
+Number of unique utterance_ids: 2.
+Bucket: bucket_0, total unique problem utterances: 1 (50.0% of all), one annotator: 1 (100.0%), two_annotators: 0 (0.0%), three+ annotators: 0 (0.0%)
+Bucket: bucket_4, total unique problem utterances: 1 (50.0% of all), one annotator: 0 (0.0%), two_annotators: 1 (100.0%), three+ annotators: 0 (0.0%)
+Bucket: none_all_good, total unique problem utterances: 1 (50.0% of all), one annotator: 1 (100.0%), two_annotators: 0 (0.0%), three+ annotators: 0 (0.0%)
+Bucket: any_problem, total unique problem utterances: 2 (100.0% of all), one annotator: 1 (50.0%), two_annotators: 1 (50.0%), three+ annotators: 0 (0.0%)
+Got 4 utterances with gold annotations. Found 8 utterances matching gold annotations from DataFrame.
+Average agreement with 4 total gold utterances annotated was:
+bucket_0: 91.7% (0 gold problem samples)
+bucket_1: 100.0% (1 gold problem samples)
+bucket_2: 75.0% (0 gold problem samples)
+bucket_3: 91.7% (0 gold problem samples)
+bucket_4: 100.0% (2 gold problem samples)
+none_all_good: 58.3% (1 gold problem samples)
+Average agreement problem samples only with 4 total gold utterances annotated was:
+bucket_0: nan% (0 gold problem samples)
+bucket_1: 100.0% (1 gold problem samples)
+bucket_2: nan% (0 gold problem samples)
+bucket_3: nan% (0 gold problem samples)
+bucket_4: 100.0% (2 gold problem samples)
+none_all_good: 33.3% (1 gold problem samples)
+Calculating agreement on 8 annotations.
+Fleiss' kappa for bucket_0 is: -0.410.
```

Comment:
somehow I'm confused at the calculation of fleiss' kappa. I thought the fleiss kappa is calculated the agreement among the same set of raters. Given how the turn annotation task is set up, each assignment can have different set of raters, so I'm expecting to see fless kappa calculated PER assignment PER bucket?

Reply:
Hmm interesting - I've read over the definition of Fleiss' kappa at https://en.wikipedia.org/wiki/Fleiss%27_kappa and it looks like by default it sums over assignments (which I think this is what they call "subjects" in this article), but I agree that it might be useful to break this metric out by assignment as well. I've added a TODO line for that in this script, as well as a note in my list of future improvements.